advanced/matrix/matrix.py

```diff
@@ -0,0 +1,328 @@
+        second_tup = ()
+        for i in range(len_tup):
+            second_tup += (1,)
+
+        for j in range(len_tup):
+            mat += (second_tup,)
+
+        return Matrix(mat)
+
+    def __mul__(self, other):
+        """
+        Method that implement a Matrix multiplication. In this format: self * other.
+        :param self: The first Matrix object.
+        :param other: A second object.
+        :return: New Matrix object which is the matrix multiplication result of self and other.
+        """
+        # In this format: Matrix * scalar
+        if type(other) == int or type(other) == float:
+            return self._scalar_multiplication(other)
+
+        if type(other) == Matrix:
+            return self.matrix_multiplication(other)
+
+        else:
+            raise TypeError("Type error raised, can't multiply Matrix object by '%s' object."
+                            % other.__class__.__name__)
+
+    def __rmul__(self, other):
+        """
+        Method that implement a Matrix multiplication by scalar in this format: scalar * Matrix. (self * other)
+        :param other: Int or float value.
+        :return:
+        """
+        return self.__mul__(other)
+
+    def _scalar_multiplication(self, scalar):
+        """
+        Method that implement a Matrix and scalar multiplication .
+        :param self: The first Matrix object.
+        :param scalar: The scalar. (Int or float object).
+        :return: New Matrix object which is the matrix multiplication result of the Matrix object given and the scalar.
+        """
+        mat_result = ()
+        for i in range(len(self._matrix)):
+            mat_result += (tuple(map((lambda x: x * scalar), self.tuples()[i])),)
+
+        return Matrix(mat_result)
+
+    def matrix_multiplication(self, other):
+        """
+        Method that implement a Matrix multiplication.
+        :param self: The first Matrix object.
+        :param other: The second Matrix object.
+        :return: New Matrix object which is the matrix multiplication result of self and other.
+        """
+        if type(other) != Matrix:
+            raise TypeError("Matrix multiplication have to be with 2 Matrix objects")
+
+        # Changing the second Matrix so the rows and columns would switch.
+        # This way implementing the multiplication formula it is more simple.
+        other = other.opposite_matrix()
+
+        mat_len = len(self._matrix)
+        ans_mat = ()
+
+        # col_ind refers to the index of a tuple in the first Matrix given.
+        for mat1_ind in range(mat_len):
+            second_tup = ()
+
+            # row_ind refers to the index of a tuple in the second Matrix (The opposite one)).
+            for mat2_ind in range(mat_len):
+                # Adding the multiplication result of every value in the 2 tuples with the same index.
+                second_tup += (sum(list(map((lambda x, y: x * y), self.tuples()[mat1_ind], other.tuples()[mat2_ind]))),)
```

Comment:
Matrix multiplication is not item-by-item, it's more complicated.

Reply:
I checked it with a matrix calculator on a website, it worked every time.